### PR TITLE
Pixel Triplet/Quintuplets Kernels to Alpaka

### PR DESCRIPTION
**Pull request by GNiendorf**
Moves the two kernels in PixelTriplet.cu to Alpaka, along with their corresponding device functions.

This PR Timing
Screenshot 2023-03-24 at 4 20 56 PM

Master Timing
Screenshot 2023-03-21 at 8 49 45 AM

Comparison Plots to Master: [Here](https://www.classe.cornell.edu/~gsn27/www/www/PR259/mywork_a874fa-PU200_888722-PU200/summary/)

### CODE REVIEW

**Comment by GNiendorf** [comment]
Moved hypotf to alpaka in the last commit. Small change, but here are the new validation plots - [Here](https://www.classe.cornell.edu/~gsn27/www/www/PR259/mywork_345a9e-PU200_888722-PU200/summary/)

**Review by VourMa — COMMENTED** [review]
The PR looks good, just a few minor comments before merging.

I have a general question, definitely not to be addressed in this PR but possibly to take a note of:
@GNiendorf Did you also notice a lot of repeating code for the pT3 and pT5 creation? Do you think that this part of the code could be merged/simplified at some point in the future?

**Comment by GNiendorf** [comment]
> The PR looks good, just a few minor comments before merging.
> 
> I have a general question, definitely not to be addressed in this PR but possibly to take a note of: @GNiendorf Did you also notice a lot of repeating code for the pT3 and pT5 creation? Do you think that this part of the code could be merged/simplified at some point in the future?

Yup I noticed that as well. This code should definitely be simplified in the future, I agree.

**Comment by VourMa** [comment]
> > I have a general question, definitely not to be addressed in this PR but possibly to take a note of: @GNiendorf Did you also notice a lot of repeating code for the pT3 and pT5 creation? Do you think that this part of the code could be merged/simplified at some point in the future?
> 
> Yup I noticed that as well. This code should definitely be simplified in the future, I agree.

Ok, I am making an issue out of that.

File: SDL/PixelTriplet.cu [hunk 1]
```diff
@@ -86,1393 +85,57 @@ void SDL::pixelTriplets::resetMemory(unsigned int maxPixelTriplets,cudaStream_t
...
-
-    else if(layer1 == 7 and layer2 == 8 and layer3 == 9)
-    {
-        return rzChiSquared < 3.9263f; //CMSSW 12 geo, add charge, new pixel map
-    }
-/*    else if(layer1 == 7 and layer2 == 8 and layer3 == 14)
-    {   
-        return rzChiSquared < 16.836f; //CMSSW 12 geo, add charge, new pixel map
-    }
-*/
```
[the remark this answers is not in the document]
Comment (VourMa) [reply SDL/PixelTriplet.cu:429]: Thanks for the reply. Then, propose to include these lines in the following way:
```c++
/*
    else if(layer1 == 7 and layer2 == 8 and layer3 == 14)
    {   
        // PS+PS+2S in endcap layers 1+2+3, which is not really feasible in the current geometry, without skipping barrel layers 1 and 2 (not allowed by algorithm logic).
    }
*/
```

[the remark this answers is not in the document]
Comment (VourMa) [reply SDL/PixelTriplet.cu:429]: @YonsiG However, I see the same category commented-in here:
https://github.com/SegmentLinking/TrackLooper/blob/master/SDL/PixelTriplet.cu#L350-L353
So, what's the deal? How should we handle this category?

[the remark this answers is not in the document]
Comment (GNiendorf) [reply SDL/PixelTriplet.cu:429]: Also here: https://github.com/SegmentLinking/TrackLooper/blob/398270ac8b1d6b6e634d2fbcfe97b9aef9566c1d/SDL/PixelTriplet.cu#L664-L667

[the remark this answers is not in the document]
Comment (YonsiG) [reply SDL/PixelTriplet.cu:429]: Yes, they are the same regions. I raised this question when I did the rzchi2 study, checked and made sure we don't have pT3 in this region, so I comment it out this region in rzchi2. However, the rphi chi2 and chi2 cuts regions were older versions(maybe older geometry or something), and I didn't touch them when I tune the rzchi2. My suppose is that these cuts in none existing regions don't do anything, but nobody has checked that. We can leave it there as it is and when people retuning this part, they can optimize and remove redundant regions

**Review by VourMa — APPROVED** [review]
All comments addressed - thanks! Merging...

**Pull request merged**